src/api/v4/dataview/time-series/index.js

```diff
@@ -0,0 +1,177 @@
+var _ = require('underscore');
+var Base = require('../base');
+var HistogramDataviewModel = require('../../../../dataviews/histogram-dataview-model');
+var parseTimeSeriesData = require('./parse-time-series-data');
+var timeAggregation = require('../../constants').timeAggregation;
+var isValidTimeAggregation = require('../../constants').isValidTimeAggregation;
+
+function hoursToSeconds (hours) {
+  return hours * 3600;
+}
+
+/**
+ * Time-Series dataview object
+ *
+ * @param {carto.source.Base} source - The source where the dataview will fetch the data.
+ * @param {string} column - The column name to get the data.
+ * @param {object} options
+ * @param {carto.dataview.timeAggregation} [options.aggregation=auto] - Granularity of time aggregation.
+ * @param {number} offset - Amount of hours to displace the aggregation from UTC.
+ * @param {boolean} localTimezone - Indicates to use the user local timezone or not.
+ * @constructor
+ * @extends carto.dataview.Base
+ * @memberof carto.dataview
+ * @api
+ */
+function TimeSeries (source, column, options) {
+  this._initialize(source, column, options);
+  this._aggregation = this._options.aggregation;
+  this._offset = this._options.offset;
+  this._localTimezone = this._options.localTimezone;
+}
+
+TimeSeries.prototype = Object.create(Base.prototype);
+
+TimeSeries.prototype.DEFAULTS = {
+  aggregation: timeAggregation.AUTO,
+  offset: 0,
+  localTimezone: false
+};
+
+/**
+ * Return the resulting data
+ *
+ * @return {TimeSeriesData}
+ * @api
+ */
+TimeSeries.prototype.getData = function () {
+  if (this._internalModel) {
+    return parseTimeSeriesData(this._internalModel.get('data'), this._internalModel.get('nulls'), this._internalModel.get('totalAmount'), this._internalModel.getCurrentOffset());
+  }
+  return null;
+};
+
+/**
+ * Set time aggregation
+ * 
+ * @param {carto.dataview.timeAggregation} aggregation
+ * @return {carto.dataview.TimeSeries} this
+ * @api
+ */
+TimeSeries.prototype.setAggregation = function (aggregation) {
+  this._validateAggregation(aggregation);
+  this._changeProperty('aggregation', aggregation);
+  return this;
+};
+
+/**
+ * Return the current time aggregation
+ * 
+ * @return {carto.dataview.timeAggregation} Current time aggregation
+ * @api 
+ */
+TimeSeries.prototype.getAggregation = function () {
+  return this._aggregation;
+};
+
+/**
+ * Set time offset
+ * 
+ * @param {number} offset
+ * @return {carto.dataview.TimeSeries} this
+ * @api
+ */
+TimeSeries.prototype.setOffset = function (offset) {
+  this._validateOffset(offset);
+  var prevOffset = this._offset;
+  this._offset = offset;
+  if (this._internalModel) {
+    this._internalModel.set('offset', hoursToSeconds(offset));
+  } else if (prevOffset !== offset) {
```

Comment:
Interesting! Not sure we're doing this elsewhere but its a good thing! cc: @IagoLast @Jesus89